flask_mailgun.py

```diff
@@ -73,18 +97,72 @@ def init_app(self, app):
         this, it is an email address where messages will be sent.
         Optionally, you can configure the sender address as well
         with the MAILGUN_LOGGING_SENDER option.
+
+        There are also several template strings that can be optionally
+        configured. If not configured, they fall back to default
+        values. These options are:
+          - MAILGUN_API_URL_TEMPLATE
+          - MAILGUN_DEBUG_TEMPLATE
+          - MAILGUN_LOGGING_TEMPLATE
+          - MAILGUN_ERROR_SUBJECT_WITHEXCINFO_TEMPLATE
+          - MAILGUN_ERROR_SUBJECT_WITHOUTEXCINFO_TEMPLATE
+          - MAILGUN_ERROR_TEMPLATE
+          - MAILGUN_API_ERROR_TEMPLATE
         """
         app.extensions['mailgun'] = self
         self.debug = app.debug
+
+        self.domain = app.config['MAILGUN_DOMAIN']
+        self.key = app.config['MAILGUN_KEY']
+
+        self.api_url_template = app.config.get(
```

Comment:
I think the more Pythonic way to do this is the following. This also applies to all instances of the pattern below. It makes the code shorter with fewer conditionals, and that's always nice.

```python
self.api_url_template = app.config.get('MAILGUN_API_URL_TEMPLATE', default_api_url_template)
```